### ISSUE TITLE
[BITV] No heading for settings pages

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#35626]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

If I open the settings page of Nextcloud there is no h1 heading found in the page's source.

### Steps to reproduce

1. Open `/index.php/settings/user` or `/index.php/settings/admin`
2. Navigate through the pages from the left sidebar

### Expected behavior

A h1 in the content area for the currently selected entry, e.g. "Personal info".

The heading should have tabindex="-1" so the focus jumps to it after selecting the page.

### Installation method

Official All-in-One appliance

### Operating system

RHEL/CentOS

### PHP engine version

PHP 8.1

### Web server

Other

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
N/a
```

**Comment by ChristophWurst** [comment]
![Bildschirmfoto vom 2022-12-14 16-47-14](https://user-images.githubusercontent.com/1374172/207642296-f2a41ce2-298d-49ba-ac81-dfeb5c47f598.png)

there is already an h1 and the sections on the settings pages use h2

**Comment by michaelnissenbaum** [comment]
From accessibility perspective it doesn't make any difference whether heading is hidden or visible. Important is whether the structure of the headings makes sense in regard to the content of the page.

**Comment by nimishavijay** [comment]
> I would propose to drop the hidden h1 for this page and instead add a visible h1 with the settings page name on top of the h2 sections.

So for eg. when you are in the Security section, the top of the content will have an h1 "Security", and h2 "Two factor authentication", "Passwordless authentication", "End-to-end encryption" etc, did I understand that correctly? That sounds good to me.

**Comment by ChristophWurst** [comment]
Yes, that is how it would render

**Comment by JuliaKirschenheuter** [comment]
Headings have to be fixed in:

- [ ] Personal + Administration settings
- [ ] Activity App
- [ ] Users
- [ ] Apps
- [ ] Photos App

**Comment by nickvergessen** [comment]
> Personal + Administration settings

Most of them use the settings component we have and that has headers in them 🤔

**Comment by JuliaKirschenheuter** [comment]
> Most of them use the settings component we have and that has headers in them 

I understand what you mean. I've understood this ticket this way: somehow it should be possible to create a individual name for each page from the sidebar. For now there only headings for for example "Version", "Update" and so on available. But how is it possible to identify "Overview" which is inside of "Administration"? 

![Screenshot from 2023-02-10 16-07-45](https://user-images.githubusercontent.com/6078378/218125904-18dc25bd-5f2b-4add-a4e3-8ce0212098b7.png)

**Comment by JuliaKirschenheuter** [comment]
There is one acceptable solution which will set right `<h1>` from page to page and won't break existing page layout: existing `<h1>` have to be replaced with current value from `$_['pageTitle']`. Because of it it is important to set an initial value for `pageTitle` in each app: Users, Apps, Activity, Photos, Personal/Admin settings and Files. This is already done via setting initial value for additional param `pageTitle` in `TemplateResponse` of each responsible Controller:
- Personal/Admin settings: https://github.com/nextcloud/server/pull/29881
- Apps and Users: https://github.com/nextcloud/server/pull/36640
- Photos: https://github.com/nextcloud/photos/pull/1634
- Activity: https://github.com/nextcloud/activity/pull/1081

TODO: 
- Files: set initial value for additional param `pageTitle` in `TemplateResponse` 

In addition to initial value of `pageTitle` it is need to update it relating to current page. In all apps besides Personal/Admin settings browser history API takes place. Because of this a function which implements an update of `pageTitle` should be written in https://github.com/nextcloud/nextcloud-router. This function should be called: 
- [ ] https://github.com/nextcloud/server/blob/d33434237dc5e6f9a8515d02574adc4f5a4c0844/apps/settings/src/router.js#L129
- [ ] https://github.com/nextcloud/activity/blob/ecc10d541871ca34b9e69664600433c511f7589f/js/script.js#L62
- [ ] https://github.com/nextcloud/photos/blob/9f1351106d7c0da773426f5ee79bd9c4678376ee/src/router/index.js#L303
- [ ] Files
